pkg/grafana/synthetic-monitoring.go

```diff
@@ -28,331 +28,236 @@ import (
+		return fmt.Errorf("input file is invalid: %v", err)
 	}
-	metricsInstanceID, err := strconv.Atoi(os.Getenv("GRAFANA_SM_METRICS_ID"))
+	_, err = smClient.AddCheck(ctx, theCheck)
 	if err != nil {
-		return "", fmt.Errorf("GRAFANA_SM_METRICS_ID environment variable must be set.")
+		return err
 	}
-	logsInstanceID, err := strconv.Atoi(os.Getenv("GRAFANA_SM_LOGS_ID"))
+	return nil
+}
+
+func updateCheck(resource grizzly.Resource) error {
+	smClient, err := NewSynthenticMonitoringClient()
 	if err != nil {
-		return "", fmt.Errorf("GRAFANA_SM_LOGS_ID environment variable must be set.")
+		return err
 	}
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
 
-	type AuthRequest struct {
-		StackID           int `json:"stackId"`
-		MetricsInstanceID int `json:"metricsInstanceId"`
-		LogsInstanceID    int `json:"logsInstanceId"`
-	}
+	convertProbeNameToID(&resource)
 
-	authRequest := AuthRequest{
-		StackID:           stackID,
-		MetricsInstanceID: metricsInstanceID,
-		LogsInstanceID:    logsInstanceID,
+	theCheck, err := SpecToCheck(&resource)
+	if err != nil {
+		return fmt.Errorf("input file is invalid: %v", err)
 	}
-
-	authRequestJSON, err := json.Marshal(authRequest)
+	_, err = smClient.UpdateCheck(ctx, theCheck)
 	if err != nil {
-		return "", err
+		return err
 	}
 
-	client, err := NewHttpClient()
+	return nil
+}
+
+func SpecToCheck(r *grizzly.Resource) (synthetic_monitoring.Check, error) {
+	var smCheck synthetic_monitoring.Check
+	data, err := json.Marshal((*r)["spec"])
 	if err != nil {
-		return "", err
+		return synthetic_monitoring.Check{}, nil
 	}
-	req, err := http.NewRequest("POST", url, bytes.NewReader(authRequestJSON))
-	req.Header.Add("Authorization", "Bearer "+apiToken)
-	req.Header.Add("Content-type", "application/json")
 
-	resp, err := client.Do(req)
+	err = json.Unmarshal(data, &smCheck)
 	if err != nil {
-		return "", err
+		return synthetic_monitoring.Check{}, nil
 	}
-	defer resp.Body.Close()
 
-	if resp.StatusCode >= 400 {
-		body, _ := ioutil.ReadAll(resp.Body)
-		return "", fmt.Errorf("%d response while authenticating: %s", resp.StatusCode, string(body))
+	return smCheck, nil
+}
+
+// Probes allows accessing Probe objects by ID and by name
+func getType(check *synthetic_monitoring.Check) string {
+	if check.Settings.Ping != nil {
+		return "ping"
 	}
-	type AuthResponse struct {
-		AccessToken string `json:"accessToken"`
+	if check.Settings.Http != nil {
+		return "http"
 	}
-	data, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return "", err
+	if check.Settings.Tcp != nil {
+		return "tcp"
 	}
-	authResponse := AuthResponse{}
-	if err := json.Unmarshal(data, &authResponse); err != nil {
-		return "", grizzly.APIErr{Err: err, Body: data}
-
+	if check.Settings.Dns != nil {
+		return "dns"
 	}
-	return authResponse.AccessToken, nil
+	return ""
+}
+
+func getUID(check *synthetic_monitoring.Check) string {
```

Comment:
Pointer not needed here.
```suggestion
func getUID(check synthetic_monitoring.Check) string {
```